demo/site/src/app/block-preview/[domain]/[language]/footer/page.tsx

```diff
@@ -1,35 +1,36 @@
 "use client";
 
-import { useBlockPreviewFetch, useIFrameBridge } from "@comet/site-nextjs";
+import { createFetchInMemoryCache, useIFrameBridge } from "@comet/site-nextjs";
 import { type FooterContentBlockData } from "@src/blocks.generated";
 import { FooterContentBlock } from "@src/layout/footer/blocks/FooterContentBlock";
 import { withBlockPreview } from "@src/util/blockPreview";
+import { createGraphQLFetch } from "@src/util/graphQLClient";
 import { recursivelyLoadBlockData } from "@src/util/recursivelyLoadBlockData";
 import { useEffect, useState } from "react";
 
+const cachingFetch = createFetchInMemoryCache(fetch);
+
 export default withBlockPreview(() => {
     const iFrameBridge = useIFrameBridge();
 
-    const { fetch, graphQLFetch } = useBlockPreviewFetch();
-
     const [blockData, setBlockData] = useState<FooterContentBlockData>();
     useEffect(() => {
         async function load() {
-            if (!graphQLFetch) return;
             if (!iFrameBridge.block) {
                 setBlockData(undefined);
                 return;
             }
+            const graphQLFetch = createGraphQLFetch({ fetch: cachingFetch });
```

Comment:
Do we have to create the fetch in the effect because of the preview params?

Reply:
yes, createGraphQLFetch is always called in components because of preview params